File: Team_Workspace/Omar_Gasser/09-Inheritance/09-Inheritance.cpp
#include <iostream>
using namespace std;

class Vehicle {
protected:
    double speed;

public:
    void setSpeed(double s) {
        speed = s;
    }

    void showSpeed() {
        cout << "Current Speed: " << speed << endl;
    }
};

class SportsCar : public Vehicle {
public:
    void turboBoost() {
        speed *= 2; 
        cout << "TURBO ACTIVATED!" << endl;
    }
};

int main() {
    SportsCar my_car;

    my_car.setSpeed(50);     // Parent method
    my_car.turboBoost();    // Child method
    my_car.showSpeed();     // Parent method

    return 0;
}

File: Team_Workspace/Shahd_Tamer/Python_Tasks/09-Inheritance.py
# The Vehicle Factory
# Scenario: You are building a simulation with different types of vehicles.
#Create a Parent Class
class Vehicle :
    # Constructor
    def __init__ (self):
        self._speed  = 0.0
#Public Method: sets speed 
    def set_speed(self, s):
        if s >=0 :
            self._speed= s 
            #print(f"Speed set to {s}")   # extra
        else :
            print("Error speed must be positive")
            return 0.0
# puplic method :  shows speed
    def show_speed (self):
        print(f"Current Speed is {self._speed}")
# create child class
class SportsCar(Vehicle):
    def turbo_boost(self) :
        self._speed = self._speed * 2 
        print(" TURBO ACTIVATED! ")

# Main Execution 
my_vehicle = SportsCar()
my_vehicle .set_speed(50)
my_vehicle.turbo_boost()
my_vehicle.show_speed()